prog.go

```diff
@@ -869,6 +869,20 @@ func (p *Program) run(opts *RunOptions) (uint32, time.Duration, error) {
 		ctxOut = make([]byte, binary.Size(opts.ContextOut))
 	}
 
+	sysCtxOut := ctxOut
+	if ctxOut != nil && p.Type() == Syscall {
+		// Linux syscall program errors on non-nil ctxOut, uses ctxIn
+		// for both input and output. Shield the user from this wart.
+		sysCtxOut = nil
+		if ctxIn == nil {
```

Comment:
Why does this set ctxOut?

Reply:
`ctxOut` is unmarshaled from later.

We need it to be the same as the syscall input (`ctxIn == ctxOut`). We can't `ctxIn = ctxOut` since it has been possibly marshaled to, therefore `ctxOut = ctxIn`.